### ISSUE TITLE
Add decorators for changing squape defined thing for own function

### ISSUE THREAD

**Issue by juhoarvid** [CyberAlpaca/squish-api-python-extension#51]
Would it be possible to include own decorator in squape to use squape contextmanager decorated functions over own function?

For instance we found reasonable on our test automation to make decorator to make possible to report whole function in section. Similarly settings changes may be needed over specific function:

Report section decorated:
```
from functools import wraps

import test
from squape import report

def main():
    test.log("Hello World")
    method_to_be_sectioned()
    complex_stuff(2, "Makkara", asdfasdf=1, randomvar="healthy")

def report_in_section(section_msg, section_description=""):
    def decorate_function_to_section(func):
        @wraps(func)
        def execute_func_in_section(*args, **kwargs):
            with report.section(section_msg, section_description):
                func(*args, **kwargs)
        return execute_func_in_section
    return decorate_function_to_section

@report_in_section("This method is executed in section", "BTW it just logs random")
def method_to_be_sectioned(repeat=2):
    for _ in range(repeat):
        test.log("This message should be inside sections")


@report_in_section("More complex method")
def complex_stuff(repeat, value, **kwargs):
    for _ in range(repeat):
        test.log(f"{value} is good")
    if 'randomvar' in kwargs:
        test.log(f"and also {kwargs['randomvar']}")
```

**Comment by Adakar** [comment]
@juhoarvid Thank you for sharing a great idea 💯 
While looking into that, I realized that squape's context managers are implemented with `@contextmanager` which internally uses [ContextDecorator](https://docs.python.org/3/library/contextlib.html#contextlib.ContextDecorator). That makes them usable also as decorators. 

```
from squape.report import section

@section("This method is executed in section", "BTW it just logs random")
def method_to_be_sectioned(repeat=2):
    for _ in range(repeat):
        test.log("This message should be inside sections")
```

We should update the documentation! 😉

In case of sections, the current implementation does not support using function parameters as part of the decorator parameters.. Unfortunately, the below example won't work.
```python
@section(f"Add {name}")
def add_person(name):
    ...
```
I think it would be great to handle such cases as well - for sure something worth investigating.

**Comment by Adakar** [comment]
Sections can be used as decorators and as context managers. The documentation has been updated.
Fixed in version [v1.1](https://github.com/CyberAlpaca/squish-api-python-extension/releases/tag/v1.1)
